pkg/scheduler/utilities_test.go

```diff
@@ -376,9 +587,77 @@ func getTestUserGroup() security.UserGroup {
-	userResource := ugm.GetUserResources(userGroup)
-	groupResource := ugm.GetGroupResources(userGroup.Groups[0])
+	expectedQueuesMaxLimits := make(map[string]map[string]interface{})
+	expectedQueuesMaxLimits["root"] = make(map[string]interface{})
+	expectedQueuesMaxLimits["root.default"] = make(map[string]interface{})
+	expectedQueuesMaxLimits["root"][maxresources] = resources.NewResourceFromMap(map[string]resources.Quantity{"memory": 10, "vcores": 10})
+	expectedQueuesMaxLimits["root.default"][maxresources] = resources.NewResourceFromMap(map[string]resources.Quantity{"memory": 5, "vcores": 5})
+	expectedQueuesMaxLimits["root"][maxapplications] = uint64(2)
+	expectedQueuesMaxLimits["root.default"][maxapplications] = uint64(1)
+	internalAssertUserGroupResource(t, userGroup, expected, expectedQueuesMaxLimits)
+}
+
+func internalAssertUserGroupResource(t *testing.T, userGroup security.UserGroup, expected *resources.Resource, expectedQueuesMaxLimits map[string]map[string]interface{}) {
+	manager := ugm.GetUserManager()
+	userResource := manager.GetUserResources(userGroup)
+	groupResource := manager.GetGroupResources(userGroup.Groups[0])
+	ut := manager.GetUserTracker(userGroup.User)
+	if ut != nil {
+		maxResources := make(map[string]*resources.Resource)
+		usage := ut.GetUserResourceUsageDAOInfo()
+		internalGetMaxResource(usage.Queues, maxResources)
+		for q, qMaxLimits := range expectedQueuesMaxLimits {
+			if qRes, ok := maxResources[q]; ok {
+				assert.Equal(t, resources.Equals(qRes, qMaxLimits[maxresources].(*resources.Resource)), true)
+			}
+		}
+		maxApplications := make(map[string]uint64)
+		internalGetMaxApplications(usage.Queues, maxApplications)
+		for q, qMaxLimits := range expectedQueuesMaxLimits {
+			if qApps, ok := maxApplications[q]; ok {
+				assert.Equal(t, qApps, qMaxLimits[maxapplications].(uint64), "queue path is "+q+" actual: "+strconv.Itoa(int(qApps))+", expected: "+strconv.Itoa(int(qMaxLimits[maxapplications].(uint64))))
+			}
+		}
+	}
+
+	gt := manager.GetUserTracker(userGroup.User)
+	if gt != nil {
+		gMaxResources := make(map[string]*resources.Resource)
+		gUsage := gt.GetUserResourceUsageDAOInfo()
+		internalGetMaxResource(gUsage.Queues, gMaxResources)
+		for q, qMaxLimits := range expectedQueuesMaxLimits {
+			if qRes, ok := gMaxResources[q]; ok {
+				assert.Equal(t, resources.Equals(qRes, qMaxLimits[maxresources].(*resources.Resource)), true)
+			}
+		}
+		gMaxApps := make(map[string]uint64)
+		internalGetMaxApplications(gUsage.Queues, gMaxApps)
+		for q, qMaxLimits := range expectedQueuesMaxLimits {
+			if qApps, ok := gMaxApps[q]; ok {
+				assert.Equal(t, qApps, qMaxLimits[maxapplications].(uint64))
+			}
+		}
+	}
 	assert.Equal(t, resources.Equals(userResource, expected), true)
 	assert.Equal(t, resources.Equals(groupResource, expected), true)
 }
+
+func internalGetMaxResource(usage *dao.ResourceUsageDAOInfo, maxResources map[string]*resources.Resource) map[string]*resources.Resource {
```

Comment:
Nit: just call it `getMaxResource()`, it doesn't collide with an existing function.